src/System.Collections/src/System/Collections/Generic/SortedSet.TreeSubSet.cs

```diff
@@ -300,21 +304,35 @@ internal override int InternalIndexOf(T item)
 
             /// <summary>
             /// Checks whether this subset is out of date, and updates it if necessary.
+            /// <param name="updateCount">Updates the count variable if necessary.</param>
             /// </summary>
-            internal override void VersionCheck() => VersionCheckImpl();
+            internal override void VersionCheck(bool updateCount = false) => VersionCheckImpl(updateCount);
 
-            private void VersionCheckImpl()
+            private void VersionCheckImpl(bool updateCount)
             {
                 Debug.Assert(_underlying != null);
                 if (version != _underlying.version)
                 {
                     root = _underlying.FindRange(_min, _max, _lBoundActive, _uBoundActive);
                     version = _underlying.version;
-                    count = 0;
-                    InOrderTreeWalk(n => { count++; return true; });
+
+                    if (updateCount) {
+                        count = 0;
+                        InOrderTreeWalk(n => { count++; return true; });
+                        countVersion = _underlying.version;
```

Comment:
I don't see where this variable is used now. Shouldn't this be outside of the version != _underlying.version comparing countVersion to _underlying.version if updateCount is true?